src/util/element-object-generator.ts

```diff
@@ -40,4 +41,35 @@ export namespace ElementObjectGenerator {
                 return null;
         }
     };
+
+    export const duplicate = (element: TElement, existingIds: string[]) => {
+        let internalId: string;
+        while (!internalId || existingIds.includes(internalId)) {
+            internalId = SimpleIdGenerator.generate();
+        }
+
+        const generateId = (id: string) => {
+            switch (true) {
+                case id.includes("-copy-"): {
+                    const parts = id.split("-");
+                    const number = parseInt(parts[parts.length - 1], 10);
+                    const incrementedNumber = number + 1;
+                    const newId =
+                        parts.slice(0, -1).join("-") + "-" + incrementedNumber;
+                    return newId;
+                }
+                case id.includes("-copy"):
+                    return id + "-2";
+                default:
+                    return id + "-copy";
+            }
+        };
+
+        const baseAttributes: IBaseAttributes = {
```

Comment:
with `TElement`, you won't need to cast the output as `TElement` in the hook
```suggestion
        const duplicatedElement: TElement = {
```